syntax = "proto3";

package auth;

enum OperationResult {
  OK = 0;
  FAILED = 1;
  CANCELLED = 2;
  TIMEOUT = 3;
  OVERLOAD = 4;
};

message TaskHeader {
    repeated bytes auth_ids = 1;
};

// Inits billing and returns aut paths - also confirms auth
message BillRequest {
    repeated bytes auth_ids = 1;
    map<string, double> bill_items = 2;
};

message AuthDescr {
    bytes auth_id = 1;
    repeated string auth_keys = 2;
};

message BillResponse { repeated AuthDescr auths = 1; };

message BillAppend { map<string, double> bill_items = 1; };
message BillOverride { map<string, double> bill_items = 1; };

// send BillFinish to confirm billing operation finish
message BillFinish {
    OperationResult result = 1;
};
